src/middleware/updateValidation.js

```diff
@@ -0,0 +1,82 @@
+const bcrypt = require('bcrypt');
+const constants = require('../lib/constants');
+const UserModel = require('../models/user');
+
+// eslint-disable-next-line consistent-return
+const validateUpdate = (req, res, next) => {
+  const {
+    password,
+    passwordConfirmation,
+    address,
+    age,
+    phoneNumber,
+    username,
+  } = req.body;
+  const errorsArray = [];
+  if (username) {
+    if (!constants.USERNAME_REGEX.test(username)) {
+      errorsArray.push(constants.USERNAME_ERROR);
+    }
+  }
+  if (phoneNumber) {
+    if (!constants.PHONE_NUMBER_REGEX.test(phoneNumber)) {
+      errorsArray.push(constants.PHONE_NUMBER_ERROR);
+    }
+  }
+  if (address) {
+    const requiredFeilds = ['country', 'city', 'address0', 'address1', 'zip'];
```

Comment:
Some of these fields, address1 and zip, aren't required fields. We don't really need to ask for them.

Reply:
But if they're there, we need to check if the zip is given according to the `ZIP_REGEX: new RegExp(/^\d{5}(?:[-\s]\d{4})?$/)`